services/proxy/main.go

```diff
@@ -33,14 +40,39 @@ type session struct {
 	ymAddr string
 }
 
+
 // Login will authenticate the user. You can use any authentication system here
 // `state` is a struct with IP addr, etc. So we can use the YM API service in order
 // to request environment / entity services about this IP
 func (b *backend) Login(state *smtp.ConnectionState, username, password string) (smtp.Session, error) {
-	if username != "username" || password != "password" {
+
+	// Reverse DNS lookup
+	var client_ReverseDns = "unknown"
+	
+	if reverse, _ := net.LookupAddr(state.RemoteAddr.String()); len(reverse) != 0 {
```

Comment:
`Login` returns a struct and an error, so you can catch the error and return the error. 

```suggestion
	if reverse, err := net.LookupAddr(state.RemoteAddr.String()); err != nil {
	        return nil, err
	}
```